### PR TITLE
Added sampling for tracing

### LINKED ISSUE #1058: Traces can have configurable sampling

### PR DESCRIPTION
**Pull request by coolwednesday**
## Pull Request Template

- Closes #1058 

**Checklist:**

-   [x] I have formatted my code using  `goimport`  and  `golangci-lint`.
-   [x] All new code is covered by unit tests.
-   [x] This PR does not decrease the overall code coverage.
-   [x] I have reviewed the code comments and documentation for clarity.

**Thank you for your contribution!**

### CODE REVIEW

File: pkg/gofr/gofr.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ const (
 	shutDownTimeout        = 30 * time.Second
 	gofrTraceExporter      = "gofr"
 	gofrTracerURL          = "https://tracer.gofr.dev"
+	defaultTraceRatio      = 1
```
Comment (vipul-rawat) [inline pkg/gofr/gofr.go:42]: I don't see this constant being used anywhere.

Comment (coolwednesday) [reply pkg/gofr/gofr.go:42]: Ahh yes. I will remove it.

**Comment by Umang01-hash** [comment]
@coolwednesday Please add the newly introduced config `TRACER_RATIO` in documentation

**Review by Umang01-hash — DISMISSED** [review]

**Review by aryanmehrotra — CHANGES_REQUESTED** [review]
Configs needs to be added in configs.md file

**Pull request merged**